compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -64,19 +64,24 @@ class Driver {
 
   protected def sourcesRequired: Boolean = true
 
-  def setup(args: Array[String], rootCtx: Context): (List[AbstractFile], Context) = {
+  protected def command: CompilerCommand = ScalacCommand
+
+  def setup(args: Array[String], rootCtx: Context): (Option[List[AbstractFile]], Context) = {
     val ictx = rootCtx.fresh
-    val summary = CompilerCommand.distill(args, config.ScalaSettings(), ictx.settingsState)
+    val settings = config.ScalaSettings()
+    val summary = command.distill(args, settings, settings.defaultState)
     ictx.setSettings(summary.sstate)
     MacroClassLoader.init(ictx)
     Positioned.init(using ictx)
 
     inContext(ictx) {
       if !ctx.settings.YdropComments.value || ctx.mode.is(Mode.ReadComments) then
         ictx.setProperty(ContextDoc, new ContextDocstrings)
-      val fileNames = CompilerCommand.checkUsage(summary, sourcesRequired)(using ctx.settings)(using ctx.settingsState)
-      val files = fileNames.map(ctx.getFile)
-      (files, fromTastySetup(files))
+      val fileNamesOrNone = command.checkUsage(summary, sourcesRequired)(using ctx.settings)(using ctx.settingsState)
+      fileNamesOrNone.fold((None, ictx)) { fileNames =>
```

Comment:
Not very important but we rarely use fold on Option in the compiler, instead we prefer to match on Some(...) and None

Reply:
Fill fix